File: go-danmu/model/Collection.go
package model

import "gorm.io/gorm"

//收藏夹
type Collection struct {
	gorm.Model
	Uid   uint   `gorm:"not null"`           //所属用户
	Name  string `gorm:"type:varchar(20);"`  //收藏夹名称
	Desc  string `gorm:"type:varchar(150);"` //简介
	Cover string `gorm:"size:255;"`          //封面
	Open  bool   `gorm:"default:false"`      //是否公开
}

File: vue-danmu/src/hooks/collection.js
import { ref } from 'vue';
import { useNotification } from 'naive-ui';
import { getCollectionListAPI, addCollectionAPI } from '@/api/collect';

export default function useCollection() {
    const collections = ref([]);
    const notification = useNotification();//通知

    //获取收藏夹列表
    const getCollectionList = () => {
        getCollectionListAPI().then((res) => {
            if (res.data.code === 2000) {
                collections.value = res.data.data.collections;
            }
        })
    }

    //创建收藏夹
    const createCollection = (name) => {
        addCollectionAPI(name).then((res) => {
            if (res.data.code === 2000) {
                collections.value.push({
                    id: res.data.data.id,
                    cover: cover,
                    name: name,
                    desc: desc,
                    open: open,
                })
            }
        }).catch((err) => {
            notification.error({
                title: '创建失败',
                content: "原因:" + err.response.data.msg,
                duration: 5000,
            })
        });
    }

    return {
        collections,
        getCollectionList,
        createCollection
    }
}

